### PR TITLE
Test-driven development: Testing Your Ruby Code

### PR DESCRIPTION
**Pull request by Ceci007**
## Test-driven development: Testing Your Ruby Code 
Created `player_spec.rb` and `board_spec.rb` and implemented TDD practices.
In this Project we:
- Installed rspec gem and simplecov.
- Add Tests for Player class.
- Add Tests for Board class.

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      expect(board.winner).to eql(false)
+    end
+  end
+
+  describe %(#empty_position?) do
+    it %(returns true if the position is empty) do
+      board.move(0, 0, ' ')
+      xy_pos = 1
+      expect(board.empty_position?(xy_pos[0], xy_pos[1])).to eql(true)
+    end
```
Comment (bertil291utn) [inline spec/board_spec.rb:89]: ## Negative scenario

- Specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
Example: there is a spec for correct usage of the method [e.g. it "creates an object if all required attributes are passed"]
and also there is spec for the wrong usage of a method  [e.g. it "does not create the object if required attributes are missing"].

  - [ ] Thus, please add negative scenarios

Example:

```ruby
#positive scenario
  it %(returns true if the position is empty) do
#negative scenario
  it %(returns false if the position is not empty) do
```

File: spec/board_spec.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      board.move(1, 1, 'O')
+      board.move(2, 2, 'O')
+      expect(board.check_winner(0, 0, 'O')).not_to eql(false)
+    end
+
+    it %(Checks if there is a winner on the second diagonal) do
+      board.move(0, 2, 'X')
+      board.move(1, 1, 'X')
+      board.move(2, 0, 'X')
+      expect(board.check_winner(0, 2, 'X')).not_to eql(false)
```
Comment (bertil291utn) [inline spec/board_spec.rb:63]: ## Negative scenario

- Specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
Example: there is a spec for correct usage of the method [e.g. it "creates an object if all required attributes are passed"]
and also there is spec for the wrong usage of a method  [e.g. it "does not create the object if required attributes are missing"].

  - [ ] Thus, please add negative scenarios

Example:

```ruby
#positive scenario
it %(Checks if there is a winner ) do
#negative scenario
it %(Checks if there is no a winner) do
```

File: spec/player_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    it %(returns the player's name) do
+      expect(player_x.name).to eql('Ceci')
+    end
+
+    it %(retuns true if the position is not yet taken by player_x) do
+      expect(player_x.check_position?(1, board)).to eql(true)
+    end
+
```
Comment (bertil291utn) [inline spec/player_spec.rb:27]: ## Negative scenario

- Specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
Example: there is a spec for correct usage of the method [e.g. it "creates an object if all required attributes are passed"]
and also there is spec for the wrong usage of a method  [e.g. it "does not create the object if required attributes are missing"].

  - [ ] Thus, please add negative scenarios

Example:

```ruby
#positive scenario
 it %(returns true if the position is not yet taken by player_x) do
#negative scenario
 it %(returns false if the position is already taken by player_x) do
```

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow you have a professional README file. 

However, in order to accomplish all the requirements, you required to do these changes, listed below:
- [ ] Please add instructions about the test in the README file , a little description of the tool (rspec) you used, and how to run 
Example:
```markdown
## Tests
### Rspec
Rspec is .......
### How to run
- Here explain how to run 
```


### Summary
- Add also negative scenarios for some methods
- Add instructions on how to run, on README file 
-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”, you required to do these changes, listed below:

**Comment by marcoshdezcam** [comment]
## TSE required changes applied

- README.md updated with screenshots and instructions to run the tests
- We have added the required negative scenarios for the Board and Player classes

**Review by BrittanyBlake — APPROVED** [review]
### Status: Approved ✅ 
****
### Hi Team! 👋 
### Great job implementing the requested changes!! 🎖️ It's a 👍 from me! Go ahead and merge this pull request and keep up the good work 😄
****

 Good Luck and Happy Coding! 🍀 👩🏾‍💻

**Pull request merged**